google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1beta2/StreamWriterV2.java

```diff
@@ -336,15 +391,33 @@ private void doneCallback(Throwable finalStatus) {
   /** A builder of {@link StreamWriterV2}s. */
   public static final class Builder {
 
+    private static final long DEFAULT_MAX_INFLIGHT_REQUESTS = 1000L;
+
+    private static final long DEFAULT_MAX_INFLIGHT_BYTES = 100 * 1024 * 1024; // 100Mb.
+
     private String streamName;
 
     private BigQueryWriteClient client;
 
+    private long maxInflightRequest = DEFAULT_MAX_INFLIGHT_REQUESTS;
+
+    private long maxInflightBytes = DEFAULT_MAX_INFLIGHT_BYTES;
+
     private Builder(String streamName, BigQueryWriteClient client) {
       this.streamName = Preconditions.checkNotNull(streamName);
       this.client = Preconditions.checkNotNull(client);
     }
 
+    public Builder setMaxInflightRequests(long value) {
+      this.maxInflightRequest = value;
```

Comment:
There is a formal way of presenting this:
https://screenshot.googleplex.com/8zfp4G83oeinQwC
maybe also support throw mode? I think Bigang mentioned we'd like a block with timeout but I don't see how it can be specified with FlowControlSettings. I am fine with out it for now.

Reply:
I'm not sure if it is the formal way. According to the code comments
"/** Settings for {@link FlowController}. */"

We are not using FlowController, which is some beta feature and is subject to change based on comments, so why should we use its setting object? I would prefer to use our own inflight control here.

I don't think Bigang wants to block with timeout here. It doesn't make much sense.

Reply:
Why is this not FlowControl? It would make the new writer's interface closer to the old interface.

Reply:
Have you looked at the source code for FlowControlSettings?
/** Settings for {@link FlowController}. */

In FlowController.java:
/** Provides flow control capability. */
@BetaApi("The surface for batching is not stable yet and may change in the future.")
public class FlowController {

We are not using the CLASS named FlowController. and it is not a good idea to use a random class's setting object, because it can change anytime.

Reply:
Well, we are using BetaApi all over the place, and PubSub v1 is using this as well:
https://screenshot.googleplex.com/8KD2UYz5LETJkpi

Anyways, I am fine not to block on this. Maybe stephaniewang@ have a better insight about this.

Reply:
Yeah, since the setting is simple enough. It shouldn't be a big issue either way. I just prefer to use our own definition to avoid unexpected changes in future.

They are likely to add new fields, which we don't support. It would be confusing for our users.